dome/moving_robot.md

```diff
@@ -0,0 +1,176 @@
+
+```xml
+<!-- Moving Forward-->
+<plugin filename="libignition-gazebo-triggered-publisher-system.so"
+        name="ignition::gazebo::systems::TriggeredPublisher">
+    <input type="ignition.msgs.Int32" topic="/keyboard/keypress">
+        <match field="data">16777235</match>
+    </input>
+    <output type="ignition.msgs.Twist" topic="/cmd_vel">
+        linear: {x: 0.5}, angular: {z: 0.0}
+    </output>
+</plugin>
+```
+
+This code defines the `triggered-publisher` plugin.
+It accepts messages of type `ignition.msgs.Int32` on the `/keyboard/keypress` topic and if the value in the `data` field matches `16777235`(Up arrow key) it outputs a `Twist` message on the `cmd_vel` topic with values `x: 0.5`, `z: 0.0`.
+
+Now launch `building_robot.sdf` and our robot should move forward as we press the Up arrow key &#8593;.
+
+There is a demo explaining how the [Triggered Publisher](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo2/tutorials/triggered_publisher.md) works.
+
+### Moving using arrow keys
+
+To see what values are sent on the `/keyboard/keypress` topic when pressing the arrows we can use the `--echo` or `-e` option
+
+* Run the model in one terminal:
+
+    `ign gazebo building_robot.sdf`
+
+* In another terminal run the following command:
+
+    `ign topic -e -t /keyboard/keypress`
+
+Start pressing the arrows keys and see what values they give:
+
+* Left &#8592;  : 16777234
+* Up  &#8593;   : 16777235
+* Right &#8594; : 16777236
+* Down &#8595;  : 16777237
+
+We will add the `Triggered publisher` plugin for each arrow key.
+For example, the Down arrow:
+
+```xml
+<!-- Moving Backward-->
+<plugin filename="libignition-gazebo-triggered-publisher-system.so"
+        name="ignition::gazebo::systems::TriggeredPublisher">
+    <input type="ignition.msgs.Int32" topic="/keyboard/keypress">
+        <match field="data">16777237</match>
+    </input>
+    <output type="ignition.msgs.Twist" topic="/cmd_vel">
+        linear: {x: -0.5}, angular: {z: 0.0}
+    </output>
+</plugin>
+```
+
+Map each arrow (key stroke) with the desired message (movement) as we did with the backward arrow:
+
+* Left &#10142; 16777234 &#10142; linear: {x: 0.0}, angular: {z: 0.5}
+* Up &#10142; 16777235 &#10142; linear: {x: 0.5}, angular: {z: 0.0}
+* Right &#10142; 16777236 &#10142; linear: {x: 0.0}, angular: {z: -0.5}
+* Down &#10142; 16777237 &#10142; linear: {x: 0.5}, angular: {z: 0.0}
+
+Now it's your turn try to make the robot move using different keys.
+
+In the [next tutorial](sdf_worlds), you'll learn to create your own simulated world with SDF.
+
+## Video walk-through
+
+A video walk-through of this tutorial is available from our YouTube channel: [Ignition tutorials: Moving robot](https://youtu.be/oHtQYPDGk3Y).
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/oHtQYPDGk3Y" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

Comment:
This doesn't render well in GitHub, maybe we should remove this line since it's linked above already